### ISSUE TITLE
option boot.loader.grub.memtest86.enable not taking effect for iso image builds

### ISSUE THREAD

**Issue by cherusk** [NixOS/nixpkgs#136891]
### Describe the bug
The nixos option  boot.loader.grub.memtest86.enable is not taking effect for iso-image.nix builds. No entries for passmark memtest86 can be created. That's because the iso-image.nix assembles it's own grub2 configuration based on isolinux.

### Steps To Reproduce
1. Build the iso image along `nix-build '<nixpkgs/nixos>' -A config.system.build.isoImage -I nixos-config=configuration.nix` for following minimalistic configuration:
```
{ config, lib, pkgs, ...  }: {

  imports = [
    <nixpkgs/nixos/modules/installer/cd-dvd/installation-cd-minimal.nix>
    # Provide an initial copy of the NixOS channel so that the user
    # doesn't need to run "nix-channel --update" first.
    <nixpkgs/nixos/modules/installer/cd-dvd/channel.nix>
  ];

  boot = {
    loader = {
      efi.canTouchEfiVariables = true;
      grub.enable = true;
      # Memtest
      grub = {
        efiSupport = true; # UEFI boot necessary for passmark MemTest86
        memtest86.enable = true;
        extraEntriesBeforeNixOS = true;
      };
    };
  };
  nixpkgs.config.allowUnfree = true; # for passmark MemTest86 to be available
}
```
2. boot the resulting iso image in efi mode on some according machine
3. verfiy no passmark memtest86 boot entry available

### Expected behavior
The option boot.loader.grub.memtest86.enable works as documented for any form of bootable media created with nixos-generators.

### Additional context
Essentially, we'd only need another entry along the lines of the memtest86+ custom iso entry  replicating the behaviour of [2] added to the iso-image.nix. That has to be in EFI boot partitions, but iso-images are default EFI bootable hence that shouldn't pose a major difficulty.
[1] https://github.com/NixOS/nixpkgs/blob/7e9b0dff974c89e070da1ad85713ff3c20b0ca97/nixos/modules/installer/cd-dvd/iso-image.nix#L163
[2] https://github.com/NixOS/nixpkgs/blob/863e6e8820ac213ab1baf14d7cb408d599bd359f/nixos/modules/system/boot/loader/grub/memtest.nix#L92

### Notify maintainers
@samueldr 
@edolstra 

### Metadata
The case for latest 21.05

Maintainer information:
```yaml
# a list of nixos modules affected by the problem
module:
installer
```

**Comment by samueldr** [comment]
The NixOS iso installer does not use the modules system's GRUB generator (or any other) for the boot files.

I'm not saying your issue is invalid, I'm saying this is the reason.

To *automatically* make use of anything provided by those modules, the iso image building would need to be revamped to re-use the tooling provided in the modules system, *and* the tooling in the modules system would need to be reworked to work better for "non system" use cases.

**Comment by cherusk** [comment]
@samueldr you phrased it more accurately than I had done. :heavy_check_mark: 

> and the tooling in the modules system would need to be reworked to work better

I agree. That's quite a copious effort. Hence, for now, could we just add the passmark memtest efi binary as it's done for open memtest86+ https://github.com/NixOS/nixpkgs/blob/7e9b0dff974c89e070da1ad85713ff3c20b0ca97/nixos/modules/installer/cd-dvd/iso-image.nix#L163? Can be optional also.

**Comment by samueldr** [comment]
It would be required to be optional, as AFAIUI the passmark memtest86+ binary is not Free software, and might not be redistributable here.

**Comment by cherusk** [comment]
@samueldr
yes, that's also what they do in memtest.nix, as mentioned above:
https://github.com/NixOS/nixpkgs/blob/863e6e8820ac213ab1baf14d7cb408d599bd359f/nixos/modules/system/boot/loader/grub/memtest.nix#L25

`nixpkgs.config.allowUnfree = true;` is a must for passmark memtest to work there.

`nixpkgs.config.allowUnfree` could be well used as discriminator in iso-image.nix as well.
